modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -2604,15 +2610,15 @@ def concat(
             # frame. Typically, if we know the width for any partition in a
             # column, we know the width for the first partition in the column.
             # So just check the widths of the first row of partitions.
-            new_widths = []
-            if new_partitions.size > 0:
-                for part in new_partitions[0]:
-                    if part._width_cache is not None:
-                        new_widths.append(part.width())
-                    else:
-                        new_widths = None
-                        break
-
+            if not new_widths:
+                new_widths = []
+                if new_partitions.size > 0:
+                    for part in new_partitions[0]:
+                        if part._width_cache is not None:
+                            new_widths.append(part.width())
```

Comment:
I assume that getting `part.width()` forces us to materialize the partition which is what was causing the performance to be slower?

Reply:
In my case `_width_cache` is None, so `width` isn't called here, bit it's called in `_filter_empties`.